src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -1988,46 +1988,62 @@ private void VisitString(ParseNode node)
             AddOpcode(new OpcodePush(new StringValue(node.Token.Text.Trim('"'))));
         }
 
-        ///<summary>
-        /// Check for if the rightmost thing in the var_identifier node
-        /// is a suffix term.  i.e. return true if the var_identifier is:<br/>
-        ///    AAA:BBB, or<br/>
-        ///    AAA[0]:BBB,<br/>
-        /// but NOT if it's this:<br/>
-        ///    AAA:BBB[0].<br/>
-        /// (Which does *contain* a suffix, but not as the rightmost part of it.  The rightmost
-        /// part of it is the array indexer "[0]".)
+        /// <summary>
+        /// Given a ParseNode that should be somewhere in the Suffix term portion of a parse tree,
+        /// return the Suffix trailer or suffix term at the end of it that has the final suffix identifier.
+        /// If the term doesn't end with a suffix identifier, then this returns null.
         /// </summary>
-        private bool VarIdentifierEndsWithSuffix(ParseNode node)
+        /// <returns>The variable identifier end suffix.</returns>
+        /// <param name="node">Node.</param>
+        private ParseNode FindVarIdentifierEndNode(ParseNode node)
         {
             // If it's a var_identifier being worked on, drop down one level first
             // to get into the actual meat of the syntax tree it represents:
             if (node.Token.Type == TokenType.varidentifier)
-                return VarIdentifierEndsWithSuffix(node.Nodes.First());
+                return FindVarIdentifierEndNode(node.Nodes.First());
 
             // Descend the rightmost children until encountering the deepest node that is
             // still a suffix_trailer, array_trailer, or function_trailer.  If that node
             // was a suffix_trailer, return true, else it's false.
             ParseNode prevChild = node;
             ParseNode thisChild = node.Nodes.Last();
-            
+
             bool descendedThroughAColon = false; // eeeewwww, sounds disgusting.
-                
+
             while (thisChild.Token.Type == TokenType.suffix_trailer ||
-                   thisChild.Token.Type == TokenType.suffix || 
-                   thisChild.Token.Type == TokenType.suffixterm || 
-                   thisChild.Token.Type == TokenType.suffixterm_trailer || 
-                   thisChild.Token.Type == TokenType.array_trailer ||
-                   thisChild.Token.Type == TokenType.function_trailer)
+                thisChild.Token.Type == TokenType.suffix || 
+                thisChild.Token.Type == TokenType.suffixterm || 
+                thisChild.Token.Type == TokenType.suffixterm_trailer || 
+                thisChild.Token.Type == TokenType.array_trailer ||
+                thisChild.Token.Type == TokenType.function_trailer)
```

Comment:
These whitespace changes are from Monodevelop autoformatting things when I cut & paste to move the code.